### PR TITLE
CTest: more docs on `-R` regular expression filtering

### PR DESCRIPTION
**Pull request by EZoni**
Add docs on common use case for `-R` regular expression filtering with CTest.

### CODE REVIEW

File: Docs/source/developers/testing.rst [hunk 1]
```diff
@@ -73,13 +73,25 @@ For easier debugging, it can be convenient to run the tests on your local machin
 
   .. code-block:: sh
 
-       ctest --test-dir build -R laser_acceleration
+       ctest --test-dir build -R "laser_acceleration"
```
Comment (dpgrote) [inline Docs/source/developers/testing.rst]: I don't think the quotes are needed here. Same for the `-E` option.

Comment (ax3l) [reply Docs/source/developers/testing.rst]: Correct, only needed when `"actual.*regular\.expressions"` are used.

Comment (ax3l) [reply Docs/source/developers/testing.rst]:
```suggestion
       ctest --test-dir build -R laser_acceleration
```

File: Docs/source/developers/testing.rst [hunk 2]
```diff
@@ -73,13 +73,25 @@ For easier debugging, it can be convenient to run the tests on your local machin
...
 
 * Run tests except those matching the regular expression ``laser_acceleration``:
 
   .. code-block:: sh
 
-       ctest --test-dir build -E laser_acceleration
+       ctest --test-dir build -E "laser_acceleration"
```
Comment (ax3l) [inline Docs/source/developers/testing.rst]:
```suggestion
       ctest --test-dir build -E laser_acceleration
```

File: Docs/source/developers/testing.rst [hunk 3]
```diff
@@ -73,13 +73,25 @@ For easier debugging, it can be convenient to run the tests on your local machin
...
+
+* Sometimes two or more tests share a large number of input parameters and differ by a small set of options.
+  Such tests typically also share a base string in their names.
+  For example, you can find three different tests named ``test_3d_langmuir_multi``, ``test_3d_langmuir_multi_nodal`` and ``test_3d_langmuir_multi_picmi``.
+  In such a case, if you wish to run the test ``test_3d_langmuir_multi`` only, this can be done again with the ``-R`` regular expression filter via
```
Comment (ax3l) [inline Docs/source/developers/testing.rst]:
```suggestion
  In such a case, if you wish to run the test ``test_3d_langmuir_multi`` only, this can be done again with the ``-R`` regular `expression filter <https://regex101.com>`__ via
```

File: Docs/source/developers/testing.rst [hunk 4]
```diff
@@ -73,13 +73,25 @@ For easier debugging, it can be convenient to run the tests on your local machin
...
+
+  .. code-block:: sh
+
+       ctest --test-dir build -R "test_3d_langmuir_multi\..*"
+
+  Note that filtering with ``-R "test_3d_langmuir_multi"`` would include the other existing tests and would not be sufficient.
```
Comment (ax3l) [inline Docs/source/developers/testing.rst]:
```suggestion
  Note that filtering with ``-R "test_3d_langmuir_multi"`` would include the additional tests that have the same string in them and can be sufficient if one aims to isolate a single test.
```

File: Docs/source/developers/testing.rst (hunk 1 above)
Comment (EZoni) [reply Docs/source/developers/testing.rst]: Yes, I know it's not needed here, but I thought it was a good idea to add it here too for consistency, to sort of signal that one can use regular expressions here. I thought it could be okay to keep it in our examples in the docs.

**Review by ax3l — APPROVED** [review]

**Review by ax3l — APPROVED** [review]

File: Docs/source/developers/testing.rst [hunk 5]
```diff
@@ -81,6 +81,18 @@ For easier debugging, it can be convenient to run the tests on your local machin
...
+* Sometimes two or more tests share a large number of input parameters and differ by a small set of options.
+  Such tests typically also share a base string in their names.
+  For example, you can find three different tests named ``test_3d_langmuir_multi``, ``test_3d_langmuir_multi_nodal`` and ``test_3d_langmuir_multi_picmi``.
+  In such a case, if you wish to run the test ``test_3d_langmuir_multi`` only, this can be done again with the ``-R`` regular `expression filter <https://regex101.com>`__ via
+
+  .. code-block:: sh
+
+       ctest --test-dir build -R "test_3d_langmuir_multi\..*"
+
+  Note that filtering with ``-R "test_3d_langmuir_multi"`` would include the additional tests that have the same string in them and can be sufficient if one aims to isolate a single test.
```
Comment (EZoni) [inline Docs/source/developers/testing.rst]: Could we maybe rephrase like this?
```suggestion
  Note that filtering with ``-R "test_3d_langmuir_multi"`` would include the additional tests that have the same string in the name and can be sufficient if one aims to isolate a single test.
```

Comment (ax3l) [reply Docs/source/developers/testing.rst]: How about `their` name?

Comment (ax3l) [reply Docs/source/developers/testing.rst]: and `substring`

Comment (ax3l) [reply Docs/source/developers/testing.rst]:
```suggestion
  Note that filtering with ``-R "test_3d_langmuir_multi"`` would include the additional tests that have the same substring in their name and can be sufficient if one aims to isolate a single test.
```

Comment (EZoni) [reply Docs/source/developers/testing.rst]: Wait, this should be "would not be sufficient to isolate a single test", no?

File: Docs/source/developers/testing.rst [hunk 6]
```diff
@@ -81,6 +81,18 @@ For easier debugging, it can be convenient to run the tests on your local machin
...
+* Sometimes two or more tests share a large number of input parameters and differ by a small set of options.
+  Such tests typically also share a base string in their names.
+  For example, you can find three different tests named ``test_3d_langmuir_multi``, ``test_3d_langmuir_multi_nodal`` and ``test_3d_langmuir_multi_picmi``.
+  In such a case, if you wish to run the test ``test_3d_langmuir_multi`` only, this can be done again with the ``-R`` regular `expression filter <https://regex101.com>`__ via
+
+  .. code-block:: sh
+
+       ctest --test-dir build -R "test_3d_langmuir_multi\..*"
+
+  Note that filtering with ``-R "test_3d_langmuir_multi"`` would include the additional tests that have the same substring in their name and can be sufficient if one aims to isolate a single test.
```
Comment (EZoni) [inline Docs/source/developers/testing.rst]: That's what I meant:
```suggestion
  Note that filtering with ``-R "test_3d_langmuir_multi"`` would include the additional tests that have the same substring in their name and would not be sufficient to isolate a single test.
```

Comment (ax3l) [reply Docs/source/developers/testing.rst]: Ok, but I used could because this is just an example - not all example names are 5x extended :)

Comment (EZoni) [reply Docs/source/developers/testing.rst]: Ah I see. Here I was using the negative form because I was referring to the specific example in the text, of the three Langmuir tests with shared substring. So I meant to connect this sentence with that example, since it's in the same list item. That's why I got confused.
